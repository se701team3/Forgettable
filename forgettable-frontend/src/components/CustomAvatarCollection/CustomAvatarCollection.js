import * as React from 'react';
import {AvatarGroup, Avatar} from '@mui/material';
import classes from './CustomAvatarCollection.module.css';
import PropTypes from 'prop-types';
import {stringAvatar} from '../../functions/helper';
import {Link} from 'react-router-dom';
import {getImageSrcFromBuffer} from '../../functions/getImageSrcFromBuffer';

/*
 * Component for avatars with custom images and first names
 *
 * Author: Raina Song (rainasong)
 */
function CustomAvatarCollection(props) {
  const {persons} = props;

  return (
    <>
      {Object.values(persons).map((person, i) => {
        return (
          <div className={classes.MiniPersonProfile} key={person._id || i}>
            <Link to={{
              pathname: `/person/${person._id}`,
              state: {
                person: person,
              }}}
            style={{textDecoration: 'none'}}
            >
              <Avatar
                alt={`${person.name}`}
                {...!person.image &&
                stringAvatar(`${person.name}`)}
                src={getImageSrcFromBuffer(person.image)}
                sx={{
                  'width': '30px',
                  'height': '30px',
                }}
              />
            </Link>
            <p data-testid="name-element">{person.name}</p>
          </div>);
      })}
    </>
  );
};

CustomAvatarCollection.propTypes = {
  persons: PropTypes.arrayOf(PropTypes.shape({
    name: PropTypes.string.isRequired,
    image: PropTypes.string,
  })),
};

export default CustomAvatarCollection;
