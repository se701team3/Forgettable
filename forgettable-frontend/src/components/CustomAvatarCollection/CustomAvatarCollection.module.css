.MiniPersonProfile {
    display: flex;
    flex-direction: row;
    align-items: center;
    padding-right: 20px;
    margin-top: 0;
    margin-bottom: 0;
    height: 40px
}

.MiniPersonProfile > p {
    font-size: var(--text-medium);
    font-weight: 400;
    margin-left: 10px;
}