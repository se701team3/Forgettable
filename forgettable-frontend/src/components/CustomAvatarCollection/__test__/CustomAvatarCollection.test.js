import TestRenderer from 'react-test-renderer';
import ShallowRenderer from 'react-test-renderer/shallow';
import React from 'react';
import {render, screen} from '@testing-library/react';
import CustomAvatarCollection from '../CustomAvatarCollection';

const persons = [
  {
    first_name: 'Kent',
    last_name: 'Dodds',
    img: 'https://images.unsplash.com/photo-1646936218155-caedb4ed9bf7?ixlib=rb-1.2.1&ixid=MnwxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8&auto=format&fit=crop&w=1974&q=80',
  },
  {
    first_name: 'Jed',
    last_name: 'Watson',
    img: 'https://images.unsplash.com/photo-1600010437709-10176a56ea45?ixlib=rb-1.2.1&ixid=MnwxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8&auto=format&fit=crop&w=765&q=80',
  },
  {
    first_name: 'Marley',
    last_name: 'George',
    img: 'https://images.unsplash.com/photo-1646936218590-91d949706cb5?ixlib=rb-1.2.1&ixid=MnwxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8&auto=format&fit=crop&w=687&q=80',
  },
  {
    first_name: 'Jack',
    last_name: 'Winston',
  },
  {
    first_name: 'John',
    last_name: 'Doe',
  },
];

it('renders CustomAvatarCollection UI to test correct hierarchy', () => {
  const renderer = new ShallowRenderer();
  renderer.render(
      <CustomAvatarCollection persons={persons}/>,
  );

  const result = renderer.getRenderOutput();

  expect(result).toMatchSnapshot();
});
