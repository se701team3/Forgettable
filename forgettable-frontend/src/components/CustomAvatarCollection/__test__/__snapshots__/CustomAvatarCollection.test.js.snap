// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders CustomAvatarCollection UI to test correct hierarchy 1`] = `
<React.Fragment>
  <div
    className="MiniPersonProfile"
  >
    <Link
      style={
        Object {
          "textDecoration": "none",
        }
      }
      to={
        Object {
          "pathname": "/person/undefined",
          "state": Object {
            "person": Object {
              "img": "https://images.unsplash.com/photo-1646936218155-caedb4ed9bf7?ixlib=rb-1.2.1&ixid=MnwxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8&auto=format&fit=crop&w=1974&q=80",
              "name": "Kent Dodds",
            },
          },
        }
      }
    >
      <ForwardRef(Avatar)
        alt="Kent Dodds"
        src={null}
        sx={
          Object {
            "height": "30px",
            "width": "30px",
          }
        }
      >
        KD
      </ForwardRef(Avatar)>
    </Link>
    <p
      data-testid="name-element"
    >
      Kent Dodds
    </p>
  </div>
  <div
    className="MiniPersonProfile"
  >
    <Link
      style={
        Object {
          "textDecoration": "none",
        }
      }
      to={
        Object {
          "pathname": "/person/undefined",
          "state": Object {
            "person": Object {
              "img": "https://images.unsplash.com/photo-1600010437709-10176a56ea45?ixlib=rb-1.2.1&ixid=MnwxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8&auto=format&fit=crop&w=765&q=80",
              "name": "Jed Watson",
            },
          },
        }
      }
    >
      <ForwardRef(Avatar)
        alt="Jed Watson"
        src={null}
        sx={
          Object {
            "height": "30px",
            "width": "30px",
          }
        }
      >
        JW
      </ForwardRef(Avatar)>
    </Link>
    <p
      data-testid="name-element"
    >
      Jed Watson
    </p>
  </div>
  <div
    className="MiniPersonProfile"
  >
    <Link
      style={
        Object {
          "textDecoration": "none",
        }
      }
      to={
        Object {
          "pathname": "/person/undefined",
          "state": Object {
            "person": Object {
              "img": "https://images.unsplash.com/photo-1646936218590-91d949706cb5?ixlib=rb-1.2.1&ixid=MnwxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8&auto=format&fit=crop&w=687&q=80",
              "name": "Marley George",
            },
          },
        }
      }
    >
      <ForwardRef(Avatar)
        alt="Marley George"
        src={null}
        sx={
          Object {
            "height": "30px",
            "width": "30px",
          }
        }
      >
        MG
      </ForwardRef(Avatar)>
    </Link>
    <p
      data-testid="name-element"
    >
      Marley George
    </p>
  </div>
  <div
    className="MiniPersonProfile"
  >
    <Link
      style={
        Object {
          "textDecoration": "none",
        }
      }
      to={
        Object {
          "pathname": "/person/undefined",
          "state": Object {
            "person": Object {
              "name": "Jack Winston",
            },
          },
        }
      }
    >
      <ForwardRef(Avatar)
        alt="Jack Winston"
        src={null}
        sx={
          Object {
            "height": "30px",
            "width": "30px",
          }
        }
      >
        JW
      </ForwardRef(Avatar)>
    </Link>
    <p
      data-testid="name-element"
    >
      Jack Winston
    </p>
  </div>
  <div
    className="MiniPersonProfile"
  >
    <Link
      style={
        Object {
          "textDecoration": "none",
        }
      }
      to={
        Object {
          "pathname": "/person/undefined",
          "state": Object {
            "person": Object {
              "name": "John Doe",
            },
          },
        }
      }
    >
      <ForwardRef(Avatar)
        alt="John Doe"
        src={null}
        sx={
          Object {
            "height": "30px",
            "width": "30px",
          }
        }
      >
        JD
      </ForwardRef(Avatar)>
    </Link>
    <p
      data-testid="name-element"
    >
      John Doe
    </p>
  </div>
</React.Fragment>
`;
